src/invidious/yt_backend/extractors.cr

```diff
@@ -102,7 +102,11 @@ private module Parsers
       premium = false
 
       premiere_timestamp = item_contents.dig?("upcomingEventData", "startTime").try { |t| Time.unix(t.as_s.to_i64) }
+      author_verified_badge = item_contents["ownerBadges"]?.try do |badges_array|
+        badges_array.as_a.find(&.dig("metadataBadgeRenderer", "tooltip").as_s.== "Verified")
+      end
 
+      author_verified = (author_verified_badge && author_verified_badge.size > 0)
```

Comment:
Given how often this code is repeated, you should make it a function! You can pass to that function the `JSON::Any` object that contains the `"ownerBadges"` field, i.e `item_contents` in this context and `related` in a previous file.

Also, please put said function under `HelperExtractors`, in that same file (`yt_backend/extractors.cr`). That's where the other similar utility functions are.

Reply:
Sorry that i didnt  answer, i have currently many personal problems and i hope that i can get tomorrow back to coding!

Reply:
Yeah, sure, no problem ^^ Take care of yourself, that more important!

Reply:
Okay sorry, this took a bit longer than "tomorrow". 

I currently refactored everything except this extractor method. 
I remember that every time i tried implementing this as a method it breaks and i needed like a few exception cases where i would have written another helper method but at the end it felt like it was more code than before. Probably because this is the first time writing in crystal lang.